src/core/tsi/transport_security_interface.h

```diff
@@ -62,6 +62,15 @@ const char *tsi_result_to_string(tsi_result result);
 
 extern grpc_tracer_flag tsi_tracing_enabled;
 
+/* -- tsi_zero_copy_protector object --
+
+  This object protects and unprotects grpc slice buffers without memory copy
```

Comment:
Updated comments. Technically, it is still zero-copy. privacy-integrity just one more memory allocation :)